packages/frontend/src/providers/PinnedItemsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import { subject } from '@casl/ability';
+import { ApiError, PinnedItems } from '@lightdash/common';
+import React, { createContext, useContext } from 'react';
+import { UseMutateFunction } from 'react-query';
+import { useReorder } from '../hooks/pinning/usePinnedItems';
+import { useApp } from './AppProvider';
+
+type PinnedItemsContext = {
+    userCanManagePinnedItems: boolean;
+    reorderItems: UseMutateFunction<
+        PinnedItems,
+        ApiError,
+        PinnedItems,
+        unknown
+    >;
+};
+
+const Context = createContext<PinnedItemsContext>({
+    userCanManagePinnedItems: false,
+    reorderItems: () => {
+        throw new Error(
+            'Default value, did you wrap the component with PinnedItemsProvider?',
+        );
+    },
+});
```

Comment:
Hi @IrakliJani thanks for the suggestions! I've implemented all of them with a single observation:

Originally, not throwing an error when the component is not wrapped by `PinnedItemsProvider` made `usePinnedItemsContext` return `PinnedItemsContext | null` which affects usage on destructuring:

```
const { reorderItems } = usePinnedItemsContext();
// cannot destructure because return might be null
```

I thought it would be better to add a default value for the context and throw an error if we inadvertently call its methods without wrapping with provider instead of silently not calling it like `context?.reorderItems(...)`.
Doing this, we can destructure values and be explicit when the provider is missing.

What do you think? 🤔